app/views/pages/developer-guide/integrations/sending-custom-arguments-sendgrid.liquid

```diff
@@ -0,0 +1,255 @@
+      }
+    });
+  });
+```
+
+### Step 3: Create endpoint to receive and relay form data 
+
+Create the app endpoint that receives the form data and relays it to Sendgrid, after which it makes the Sendgrid response available to the Javascript code. This endpoint builds a JSON payload ready to be used as a Sendgrid request, based on the form data. Don't build the JSON directly in Javascript for security reasons. Execute a GraphQL mutation that sends the payload, then output the response from Sendgrid.
+
+#### views/pages/mail_send.json.liquid
+
+```liquid
+---
+slug: admin/mail_send
+format: json
+method: post
+---
+{% parse_json request_data %}
+  {
+    "personalizations": [
+      {
+        "to": [
+          {%- for email in params.to -%}
+          {
+            "email": {{ email | to_json }}
+          } {%- unless forloop.last -%}, {%- endunless -%}
+          {%- endfor -%}
+        ]
+        {%- if params.cc != blank -%}
+        ,
+        "cc": [
+          {%- for email in params.cc -%}
+          {
+            "email": {{ email | to_json }}
+          } {%- unless forloop.last -%}, {%- endunless -%}
+          {%- endfor -%}
+        ]
+        {%- endif -%}
+        {%- if params.bcc != blank -%}
+        ,
+        "bcc": [
+          {%- for email in params.bcc -%}
+          {
+            "email": {{ email | to_json }}
+          } {%- unless forloop.last -%}, {%- endunless -%}
+          {%- endfor -%}
+        ]
+        {%- endif -%}
+      }
+    ],
+    "from": {
+      "email": {{ params.from | to_json }}
+    },
+    "subject": {{ params.subject | to_json }},
+    "custom_args": {{ params.custom_args | to_json }},
+    "content": [
+      {%- if params.plain_content != blank -%}
+      {
+        "type": "text/plain",
+        "value": {{ params.plain_content | to_json }}
+      },
+      {%- endif -%}
+      {
+        "type": "text/html",
+        "value": {{ params.html_content | to_json }}
+      }
+    ]
+  }
+{% endparse_json %}
+
+{% graphql g = 'mail_send', data: request_data %}
+{{ g.api_call_send.response }}
+```
+
+### Step 4: Create GraphQL mutation 
+
+Create a GraphQL mutation. The "template" specifies a so-called "API Call Notification".
+
+#### graph_queries/mail_send.graphql
+
+```graphql
+mutation mail_send($data: HashObject) {
+  api_call_send(
+    data: $data
+    template: { name: "mail_send" }
+  ) {
+    response {
+      status
+      body
+    }
+    errors {
+      message
+    }
+  }
+}
+```
+
+### Step 5: Create API Call Notification 
+
+Create an API Call Notification. This file is a configuration file that contains API interfacing information such as the API URL, API key, and the API request body.
+It is recommended that the API key is not entered in clear text, but instead stored as a site constant, and fetched as follows:
+
+```"Authorization": "{{ 'Bearer ' | append: context.constants.sendgrid_api_key }}"```
+
+#### notifications/api_call_notifications/mail_send.liquid
+
+```liquid
```

Comment:
I dont see raw tags, so it will break... again, where is preview link for this pr?

Reply:
I fixed this.